WordPress/openverse
# Simplify Skip-to-content links

## Fixes



Fixes #2474 by @alexstine

## Description


This PR simplifies the "Skip to content" links, making them work without JavaScript.

Previously, the links used a Teleport to move the button to the top of the page. However, the teleports are really finicky, especially in SSR. When I tried migrating to Nuxt 3, teleports were causing client/server mismatch.
Besides, Openverse tries to use progressive enhancing, making  the server-rendered page without Javascript as much as possible work. With this change, "Skip to content" links should work even before the page is "hydrated", and JS starts working.

This PR adds the button at the top of the page in all layouts. This duplication should be removed after the migration to Nuxt 3, when we use `app.vue` as the top-level component.

`VLink.vue` is updated to not use a localized path for anchor links that start with `#` and link to an element on the same page as the link. Some `computed`s were unnecessary and removed.

## Testing Instructions

Open different pages in the app, and navigate to the "Skip-to-content" button using <kbd>Tab</kbd>. It should be the first button on the page. On button click, you should be taken to the "main content" of the page. The screen reader should start reading the main content.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
